src/main/scala/workflow/Transformer.scala

```diff
@@ -0,0 +1,53 @@
+package workflow
+
+import org.apache.spark.rdd.RDD
+import pipelines.Logging
+
+import scala.reflect.ClassTag
+
+/**
+ * Transformers are functions that may be applied both to single input items and to RDDs of input items.
+ * They may be chained together, along with [[Estimator]]s and [[LabelEstimator]]s, to produce complex
+ * pipelines.
+ *
+ * @tparam A input item type the transformer takes
+ * @tparam B output item type the transformer produces
+ */
+abstract class Transformer[A, B : ClassTag] extends TransformerNode[B] with Pipeline[A, B] {
+  private[workflow] override val nodes: Seq[Node] = Seq(this)
+  private[workflow] override val dataDeps: Seq[Seq[Int]] = Seq(Seq(Pipeline.SOURCE))
+  private[workflow] override val fitDeps: Seq[Seq[Int]] = Seq(Seq())
+  private[workflow] override val sink: Int = 0
+
+  /**
+   * Apply this Transformer to an RDD of input items
+   * @param in The bulk RDD input to pass into this transformer
+   * @return The bulk RDD output for the given input
+   */
+  def apply(in: RDD[A]): RDD[B] = in.map(apply)
+
+  /**
+   * Apply this Transformer to a single input item
+   * @param in  The input item to pass into this transformer
+   * @return  The output value
+   */
+  def apply(in: A): B
+
+  private[workflow] final def transform(dataDependencies: Seq[_], fitDependencies: Seq[TransformerNode[_]]): B = apply(dataDependencies.head.asInstanceOf[A])
```

Comment:
long lines -- is there a way you can ask intelli-j to fix all of these for you ?